Intro_Sort/README.md

```diff
@@ -0,0 +1,108 @@
+```
+
+## Example
+
+![Introsort](https://media.geeksforgeeks.org/wp-content/uploads/20190322164239/page-11.png)
+
+**Input:** 2, 10, 24, 2, 10, 11, 27, 4, 2, 4, 28, 16, 9, 8, 28, 10, 13, 24, 22, 28, 0, 13, 27, 13, 3, 23, 18, 22, 8, 8
+
+**Step 1:** 
+
+​	begin = 0, 	end = 29 	depthLimit = 8, 	a[pivot] = 8. 
+
+As the number of elements > 16,the pivot element is calculated based on the quicksort algorithm. Using the median-of-3 concept,the pivot element is calculated and the a[pivot]=8. The method partition() finds the correct position of the a[pivot] and helps in array split-up for further recursion. Here,the element 8 (in comment) lies in the correct position and in the further steps recursion will be carried over for the sub array before and after 8. 2,2,4,2,4,8,0,3,8,/8/,28,16,9,11,28,10,13,24,22,28,27,13,27,13,24,23,18,22,10,10
+
+
+
+**Step 2:** From the previous step, 
+
+​	begin=0, 	end=29, 	a[pivot]=8 T
+
+he recursion starts for the sub-array a[begin,pivot-1] and a[pivot+1,end] 
+
+2,2,4,2,4,8,0,3,8, 8 ,28,16,9,11,28,10,13,24,22,28,27,13,27,13,27,13,24,23,18,22,10,10 
+
+depthLimit = 7, 	depthLimit = 7, 	begin = 0,	end = 8 	begin = 10,	end = 29 
+
+number of element <= 16 Number of element > 16,
+
+hence sortDataUtil(10,29) that calculates the pivot and a[partition]=27. 
+
+Hence insertionSort (0,8)
+
+0,2,2,2,3,4,4,8,8, 8, 16,9,11,10,13,24,22,10,13,27,13,24,23,18,22,10, 27, 28,28,28
+
+
+
+**Step 3:** From the previous step, 
+
+depthLimit = 6,	begin = 10, 	end = 29 
+
+The recursion starts for the sub-array a[begin,partition-1] and a[partition+1,end]
+
+` 0,2,2,2,3,4,4,8,8,8,16,9,11,10,13,24,22,10,13,27,13,24,23,18,22,10,27,28,28,28     ` 
+
+ `sorted!                         depthLimit=6                             depthLimit=6 `
+
+Hence,no further recursion.
+
+ 	begin = 10, 	end = 25 	begin = 10, 	end = 25 
+
+As the number of elements < 16, insertionSort (10,25) 
+
+As the number of element < 16, insertionSort(27,29)
+
+` 0,2,2,2,3,4,4,8,8,8,9,10,10,10,11,13,13,13,16,18,22,22,23,24,27,27,28,28,28    ` 
+
+​                         `Sorted!                               Sorted!                    Sorted!                         `
+
+Hence, no further recursion hence no further recursion hence no further recursion.
+
+
+
+**Step 4:** 
+
+output: 0,2,2,2,3,4,4,8,8,8,9,10,10,11,13,13,13,16,18,22,22,23,24,24,27,27,28,28,28 
+
+In the above example, the depthLimit came until 6, and for bigger datasets the depthLimit might reach 0 for certain sub-array. The heapSort is carried out only for those sub-array for sorting them.
+
+Output: 0,2,2,2,3,4,4,8,8,8,9,10,10,10,11 13,13,13,16,18,22,22,23,24,24,27,27,28 28,28
+
+
+
+**Time Complexity**
+
+Average performance O(n log n) Optimal worst performance O(n log n) where, n = number of elements to be sorted.
+
+SEE ALSO:
+
+https://github.com/jainaman224/Algo_Ds_Notes/blob/master/Intro_Sort/introsort.py
```

Comment:
Please add References. You must have referred to articles and add source for images. A reference section is missing here.